GW2EIEvtcParser/EncounterLogic/Strikes/SotO/CosmicObservatory.cs

```diff
@@ -55,12 +60,90 @@ protected override CombatReplayMap GetCombatMapInternal(ParsedEvtcLog log)
                             (-1388, -779, 1991, 2610));
         }
 
+        internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog log, CombatReplay replay)
+        {
+            IReadOnlyList<AbstractCastEvent> casts = target.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd);
+
+            switch (target.ID)
+            {
+                case (int)TargetID.Dagda:
+                    var phaseBuff = target.GetBuffStatus(log, DagdaDuringPhase75_50_25, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.Value > 0).ToList();
+                    
+                    // Red AoE during 75-50-25 % phases
+                    var demonicBlasts = casts.Where(x => x.SkillId == DemonicBlast).ToList();
+                    foreach (AbstractCastEvent cast in demonicBlasts)
+                    {
+                        Segment phase = phaseBuff.Where(x => x.Start >= cast.Time).FirstOrDefault();
```

Comment:
Why the phase check?

Reply:
Because demonic blast is cased at 90% without spawning the red pushback aoe, and the buff itself isn't applied on cast time either, so i had to give it a threshold

Reply:
Could you add that fact as comment please? Also please rename "phaseBuff" to "phaseBuffs" and "phase" to "phaseBuff"